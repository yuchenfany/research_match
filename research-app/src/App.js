/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import './assets/App.css';
import {
  Routes, Route,
} from 'react-router-dom';
import React, { useState } from 'react';
import Login from './components/Login';
import ParticipantStudies from './components/ParticipantStudies';
import ResearcherHome from './components/ResearcherHome';
import Create from './components/Create';
import ParticipantProfile from './components/ParticipantProfile';
import ResearcherProfile from './components/ResearcherProfile';
import ResearcherEdit from './components/ResearcherEdit';
import ParticipantEdit from './components/ParticipantEdit';
import Type from './components/Type';
import Study from './components/Study';
import ResearcherStudy from './components/ResearcherStudy';
import Dashboard from './components/Dashboard';
import AddStudy from './components/AddStudy';
import EditStudy from './components/EditStudy';
import Messages from './components/Messages';
import DisplayStudies from './components/DisplayStudies';
import DeleteAccount from './components/DeleteAccount';
import Chat from './components/Chat';

function App() {
  const [user, setUser] = useState({
    username: 'participant',
    password: 'participantpass',
    enrolled: [],
    phys: [],
    psych: [],
    med: [],
    studies: [],
  });
  // const [deleteUser, setUserDelete] =
  // useState({ username: 'testuser', password: 'testuserpassword' });
  const [study, setStudy] = useState({
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  });
  const [status, setStatus] = useState({ isEnrolled: false });
  // for chat
  const [sender, setSender] = useState({ username: '', senderType: '' });
  const [receiver, setReceiver] = useState({ username: '' });

  return (
    <div className="App">
      <Routes>
        <Route exact path="/" element={<Login user={user} setUser={setUser} />} />
        <Route exact path="/participant-studies" element={<ParticipantStudies user={user} setUser={setUser} setStudy={setStudy} setStatus={setStatus} />} />
        <Route exact path="/researcher-home" element={<ResearcherHome user={user} setUser={setUser} setStudy={setStudy} />} />
        <Route exact path="/study/:id" element={<Study study={study} setStudy={setStudy} user={user} setUser={setUser} status={status} setStatus={setStatus} sender={sender} setSender={setSender} receiver={receiver} setReceiver={setReceiver} />} />
        <Route exact path="/researcher-study/:id" element={<ResearcherStudy study={study} setStudy={setStudy} user={user} setUser={setUser} status={status} setStatus={setStatus} />} />
        <Route exact path="/create" element={<Create user={user} setUser={setUser} />} />
        <Route exact path="/add-study" element={<AddStudy user={user} study={study} setStudy={setStudy} />} />
        <Route exact path="/edit-study" element={<EditStudy user={user} study={study} setStudy={setStudy} />} />
        <Route exact path="/type" element={<Type user={user} setUser={setUser} />} />
        <Route exact path="/participant-profile" element={<ParticipantProfile user={user} setUser={setUser} />} />
        <Route exact path="/researcher-profile" element={<ResearcherProfile user={user} setUser={setUser} />} />
        <Route exact path="/researcher-edit" element={<ResearcherEdit user={user} setUser={setUser} />} />
        <Route exact path="/participant-edit" element={<ParticipantEdit user={user} setUser={setUser} />} />
        <Route exact path="/dashboard" element={<Dashboard user={user} />} />
        <Route exact path="/delete-account" element={<DeleteAccount user={user} setUser={setUser} />} />
        <Route exact path="/participant-home" element={<DisplayStudies user={user} setUser={setUser} setStudy={setStudy} />} />
        <Route exact path="/messages" element={<Messages user={user} sender={sender} setSender={setSender} receiver={receiver} setReceiver={setReceiver} />} />
        <Route exact path="/chat" element={<Chat sender={sender} receiver={receiver} setSender={setSender} setReceiver={setReceiver} />} />
      </Routes>
    </div>
  );
}

export default App;
