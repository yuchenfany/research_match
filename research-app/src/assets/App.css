@import url('https://fonts.googleapis.com/css2?family=Overpass:wght@180&display=swap');

.App {
  text-align: center;
}

.container {
  margin: auto;
  text-align: center;
  display: flex;
  align-items: flex-end;
}

.title-header {
  font-family: Overpass;
  font-style: normal;
  font-size: 80px;
  line-height: 100px;
  display: flex;
  align-items: flex-end;
  text-align: center;
  
  color: #576074;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: white;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}