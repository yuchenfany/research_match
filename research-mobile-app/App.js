import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View, Button } from 'react-native';
import React, { useState, useEffect} from 'react';
import { NavigationContainer } from '@react-navigation/native';
import  { createNativeStackNavigator } from '@react-navigation/native-stack';
import Login from './components/Login'
import ParticipantHome from './components/ParticipantHome'
import Study from './components/Study'

/* REFERENCE CODE
import { NavigationContainer } from '@react-navigation/native';
import  { createNativeStackNavigator } from '@react-navigation/native-stack';
import Question from './Question';
import StartScreen from './StartScreen';
import EndScreen from './EndScreen';

// create a navigation stack
const Stack = createNativeStackNavigator();
export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Start">
        <Stack.Screen name="Start" component={StartScreen} />
        <Stack.Screen name="Question" component={Question} />
        <Stack.Screen name="End" component={EndScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
} */ 

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Login">
        <Stack.Screen name="Login" component={Login} />  
        <Stack.Screen name="ParticipantHome" component={ParticipantHome} />
        <Stack.Screen name="Study" component={Study} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

/*
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
}); */
