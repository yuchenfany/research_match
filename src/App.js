import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';
import React, { useState, useEffect} from 'react';

export default function App() {
  const [study, setStudy] = useState([]);

  async function getStudy() {
    const studyData = await fetch(`http://localhost:5000/study/0`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    const data = await studyData.json();
    console.log('MOBILE TEST');
    console.log(data);
    return data;
  }

  useEffect(() => {
    getStudy()
      .then(setStudy);
  }, []);

  return (
    <View style={styles.container}>
      <Text>Open up App.js to start working on your app!</Text>
      <Text>{study.title}</Text> 
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
