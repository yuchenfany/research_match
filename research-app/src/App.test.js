/**
 * @jest-environment jsdom
 */

/* eslint-disable react/jsx-filename-extension */

// import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test('snapshot test', () => {
  const component = renderer.create(
    <Router>
      <App />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
