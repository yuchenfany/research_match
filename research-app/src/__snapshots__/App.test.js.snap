// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="App"
>
  <div
    className="Login"
  >
    <p
      className="header"
    >
      Research Match
    </p>
    <form
      onSubmit={[Function]}
    >
      <label
        className="login-label"
        htmlFor="username"
      >
        <div
          className="username-wrapper"
        >
          <p
            className="field-label"
          >
            USERNAME
          </p>
          <input
            className="input-field"
            id="username"
            onChange={[Function]}
            type="text"
          />
          <p
            className="field-label"
          >
            PASSWORD
          </p>
          <input
            className="input-field"
            id="password"
            onChange={[Function]}
            type="text"
          />
          <span
            className="error-message"
          >
            
          </span>
        </div>
        <input
          className="button"
          type="submit"
          value="SUBMIT"
        />
        <div
          className="spacer"
        />
        <button
          className="link"
          onClick={[Function]}
          type="button"
        >
          New user sign-up
        </button>
      </label>
    </form>
  </div>
</div>
`;
